code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/external/ExternalServiceAuthProvider.java

```diff
@@ -322,6 +322,10 @@ public Identity getIdentity(String id, String scope) {
             ApiRequest request = ApiContext.getContext().getApiRequest();
             request.setAttribute(ServiceAuthConstants.ACCESS_TOKEN, accessToken);
             return tokenUtil.getIdentities();
+        } else if(ServiceAuthConstants.NO_IDENTITY_LOOKUP_SUPPORTED.get()) {
+            Set<Identity> identities = new HashSet<>();
+            identities.add(new Identity(account.getExternalIdType(), account.getExternalId()));
```

Comment:
not totally sure, but `getIdentity` does this check before setting identity true, do we need to do it as well? 
https://github.com/rancher/cattle/blob/682aad6e2019d1f8c45f1cf234a83a5134afd28b/code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/external/ExternalServiceAuthProvider.java#L271

Reply:
yep, should be fine to add this extra check.